streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -504,88 +438,66 @@ public void update(final Set<TopicPartition> topicPartitions, final Map<String,
 
     @Override
     public void closeAndRecycleState() {
-        prepareClose(true);
-
+        suspend();
+        prepareCommit();
```

Comment:
I think we actually do not need to commit (including write-checkpoint) when closeAndRecycle actually, and only need to suspend the task before recycle it. But this is out of the scope and we can discuss about this in another PR (cc @ableegoldman ).

Reply:
Assuming rebalancing does not happen often (in a stable deployment) it might be re-mature optimization?